beacon_node/lighthouse_network/src/rpc/handler.rs

```diff
@@ -131,6 +128,9 @@ where
 
     /// Logger for handling RPC streams
     log: slog::Logger,
+
+    /// ChainSpec for networking constants
+    chain_spec: ChainSpec,
```

Comment:
cloning a `ChainSpec` for every single handler seems unnecessary.

It would be preferable to add the relevant fields (`resp_timeout: Duration` and `max_rpc_size` )

In reality, we can go further and add just the `resp_timeout` because the `max_rpc_size` can be accessed via the `SubstreamProtocol` that the handler receives in the `new(..) ` as `listen_protocol.upgrade().max_rpc_size`